tests/src/core/geometry/testqgsgeometry.cpp

```diff
@@ -744,6 +746,48 @@ void TestQgsGeometry::splitCurve()
   QCOMPARE( p2->asWkt(), curve2 );
 }
 
+void TestQgsGeometry::fromBox3d()
+{
+  QgsBox3D box3d( 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 );
+  QgsGeometry result( QgsGeometry::fromBox3D( box3d ) );
+  QCOMPARE( result.wkbType(), Qgis::WkbType::MultiPolygonZ );
```

Comment:
The best way to do this check is to compare the obtained geometry's wkt against the expected wkt. That makes for a much more descriptive test and one which is easier to debug if things go wrong

Reply:
Done.